File: sophomore/CE3005/alg/classic/03_Fibonacci/Fibonacci.py
def fibonacci(n) :
    if (n == 1 or n == 2) :
        return 1
    else :
        a = 1
        b = 1

        for i in range(3, n + 1) :
            c = a + b
            a = b
            b = c
        
        return c


def recursiveFibonacci(n) :
    if (n == 1 or n == 2) :
        return 1
    else :
        a = recursiveFibonacci(n - 1)
        b = recursiveFibonacci(n - 2)

        return (a + b)

File: ce/ce3005/alg/classic/03_Fibonacci/Fibonacci.java
package Fibonacci;

public class Fibonacci {
    public int fibonacci(int n) {
        if (n == 1 || n == 2) {
            return 1;
        } else {
            int a = 1;
            int b = 1;
            int c = 0;

            for (int i = 3; i <= n; i++) {
                c = a + b;
                a = b;
                b = c;
            }

            return c;
        }
    }

    public int recursiveFibonacci(int n) {
        if (n == 1 || n == 2) {
            return 1;
        } else {
            int a = recursiveFibonacci(n - 1);
            int b = recursiveFibonacci(n - 2);
            
            return (a + b);
        }
    }
}
